staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go

```diff
@@ -112,9 +107,12 @@ func TrackStorageCleanup() {
 	defer cleanupLock.Unlock()
 
 	if cleanup != nil {
-		panic("Conflicting storage tracking")
+		// TODO(wojtek-t): Uncomment after addressing the
+		// double-counting issue in few aggregator integration tests.
+		// panic("Conflicting storage tracking")
```

Comment:
I suggest merging this PR with this as a "stop the bleeding" thing.

And I already have an opened WIP PR that is cleaning this up in: https://github.com/kubernetes/kubernetes/pull/109303 which we hopefully should be able to merge in 1.25 cycle.

Reply:
doesn't this mean we're double-registering and double cleaning-up? won't that make tests flake/fail if one test is cleaning up shared stuff while another test is in the middle of running?

Reply:
If we would be running multiple tests (within a single binary) at once, then yes.
In practice - we're running one at a time, and the reason it was needed was because of 4 tests that are also setting up aggregated apiserver, e.g. this one:
https://github.com/kubernetes/kubernetes/blob/master/test/integration/apiserver/apiserver_test.go#L831

- this is first setting up DefaultServer (L831)
- then setting up another apiserver (L837)

So this isn't a problem in our test setup.

Reply:
I really don't want to drop this protection unconditionally. I think it will be too easy to accidentally start two servers in parallel test runs and interfere with each other.

Reply:
I would like to get rid of it completely very soon - https://github.com/kubernetes/kubernetes/pull/109303 is almost working.
Does it make sense to treat this PR as stop the bleeding? [If #109303 seems acceptable for you that this one would be very temporary].

The alternative would be to wait for the other one (though clearly it's too late for that on in 1.24)

Reply:
FWIW - I have https://github.com/kubernetes/kubernetes/pull/109303 already working and out for review - so if this isn't urgent (and given time in the release with every single it's being more clear that it should probably wait), we should probably switch to that other PR which is solving this problem in much cleaner way.

@liggitt - WDYT?

Reply:
yeah, let's collapse onto that one.

Reply:
OK - closing this one.